src/controllers/csi/provisioner/controller.go

```diff
@@ -49,33 +53,62 @@ type OneAgentProvisioner struct {
 	apiReader    client.Reader
 	opts         dtcsi.CSIOptions
 	dtcBuildFunc dynakube.DynatraceClientFunc
-	fs           afero.Fs
 	recorder     record.EventRecorder
 	db           metadata.Access
 	path         metadata.PathResolver
+
+	mu                       *sync.Mutex
+	currentParallelDownloads int64
```

Comment:
maybe this is redundant now ? 
could just check the len of the map

Reply:
see: https://github.com/Dynatrace/dynatrace-operator/pull/656#discussion_r831995831